src/channel/rpc_0_6.ts

```diff
@@ -40,20 +40,22 @@ export class RpcChannel {
 
   public headers: object;
 
-  readonly retries: number;
-
   public requestId: number;
 
   readonly blockIdentifier: BlockIdentifier;
 
+  readonly retries: number;
+
+  readonly waitMode: boolean; // behave like web2 rpc and return when tx is processed
+
   private chainId?: StarknetChainId;
 
-  private speckVersion?: string;
+  private specVersion?: string;
 
-  readonly waitMode: Boolean; // behave like web2 rpc and return when tx is processed
+  private baseFetch: NonNullable<RpcProviderOptions['baseFetch']>;
 
   constructor(optionsOrProvider?: RpcProviderOptions) {
-    const { nodeUrl, retries, headers, blockIdentifier, chainId, waitMode } =
+    const { nodeUrl, retries, headers, blockIdentifier, chainId, waitMode, baseFetch } =
```

Comment:
@ivpavici pls extract `specVersion` and assign it to `this` to avoid the extra rpc call for each new instance

Reply:
thanks! this was addressed in this [PR](https://github.com/starknet-io/starknet.js/pull/1052) so we need to rebase and fix conflicts!